consensus/src/chained_bft/persistent_storage.rs

```diff
@@ -304,6 +304,8 @@ impl<T: Payload> PersistentStorage<T> for StorageWriteProxy {
         let (_, ledger_info, _, _) = read_client
             .update_to_latest_ledger(0, vec![])
             .expect("unable to read ledger info from storage");
+        // TODO: propagate the failure and let the caller recover or fail gracefully with decent error.
+        // panic is not an option
```

Comment:
I don't think you can recover from this, you should always have a ledger info in storage (if not the genesis one) IIUC, and so you can't start persistent storage without that info. @wqfish ?

Reply:
"or fail gracefully with decent error"